packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js

```diff
@@ -19,17 +21,95 @@ const EditorDivider = styled(Divider)`
 `;
 
 const style = {
-  fontSize: pxToRem(14),
   // The outline style is set on the wrapper with :focus-within
   outline: 'none',
 };
 
-const initialValue = [
-  {
-    type: 'paragraph',
-    children: [{ text: 'A line of text in a paragraph.' }],
-  },
-];
+const Image = ({ attributes, children, element }) => {
+  const { url, alternativeText, width, height } = element.image;
+
+  return (
+    <div {...attributes}>
+      {children}
+      <Box contentEditable={false}>
+        <img src={url} alt={alternativeText} width={width} height={height} />
+      </Box>
+    </div>
+  );
+};
+
+const Element = (props) => {
+  const { attributes, element, children } = props;
+
+  if (element.type === 'heading')
+    switch (element.level) {
+      case 1:
+        return <h1 {...attributes}>{children}</h1>;
+      case 2:
+        return <h2 {...attributes}>{children}</h2>;
+      case 3:
+        return <h3 {...attributes}>{children}</h3>;
+      case 4:
+        return <h4 {...attributes}>{children}</h4>;
+      case 5:
+        return <h5 {...attributes}>{children}</h5>;
+      case 6:
+        return <h6 {...attributes}>{children}</h6>;
+      default: // do nothing
+        return null;
+    }
+  else
+    switch (element.type) {
+      case 'link':
+        return (
+          <a href={element.url} {...attributes}>
+            {children}
+          </a>
+        );
+      case 'code':
+        return (
+          <pre {...attributes}>
+            <code>{children}</code>
+          </pre>
+        );
+      case 'quote':
+        return <blockquote {...attributes}>{children}</blockquote>;
+      case 'list':
+        if (element.format === 'ordered') return <ol {...attributes}>{children}</ol>;
+
+        return <ul {...attributes}>{children}</ul>;
+      case 'list-item':
+        return <li {...attributes}>{children}</li>;
+      case 'image':
+        return <Image {...props} />;
+      default:
+        return <p {...attributes}>{children}</p>;
+    }
+};
+
+const Leaf = ({ attributes, children, leaf }) => {
+  if (leaf.bold) {
+    children = <strong>{children}</strong>;
+  }
+
+  if (leaf.italic) {
+    children = <em>{children}</em>;
+  }
+
+  if (leaf.underline) {
+    children = <u>{children}</u>;
+  }
+
+  if (leaf.strikethrough) {
+    children = <del>{children}</del>;
+  }
+
+  if (leaf.code) {
+    children = <code>{children}</code>;
+  }
+
+  return <span {...attributes}>{children}</span>;
+};
```

Comment:
There's so much repetition here, have you looked at how this has been done in other codebases to be cleaner and a bit more concise? Payload is a nice example imo. It uses more of a factory method so you just pump out the component you need based on the name etc.

Reply:
I checked payload's code you are referring, `renderLeaf` function returns the rendered content, which can include multiple nested Leaf components if there are matched/enabled leaves or a single <span> element if there are no matches. 
For `renderElement` they do not have any method directly rendering the element. 
Also if you check each Leaf and Elements component code, its all same as above. they have just separate file for each.

IMO its adding the complexity to the code even to read it, as we do not have this functionality included yet in this task.
We can gradually change to optimise further. but for now I prefer this to be simple to read.

if we still need to optimise I can add, 
```
const RenderComponent = ({ component: Component, children, ...props }) => {
  return <Component {...props}>{children}</Component>;
};
```

Reply:
Personally I'm fine with the current approach. It is simple and readable. I'd prefer to not hold this PR up for a refactor / optimization we could do a bit further down the road or in parallel in another PR as time and bandwidth permits. What do y'all think?

Reply:
I'll create a ticket in Jira referencing this comment.

Reply:
> Personally I'm fine with the current approach. It is simple and readable. I'd prefer to not hold this PR up for a refactor / optimization we could do a bit further down the road or in parallel in another PR as time and bandwidth permits. What do y'all think?

sounds good to me 👍🏼